src/pages/challenges.tsx

```diff
@@ -23,7 +22,7 @@ const Challenges: FC = () => {
   const [filteredChallenge, setFilteredChallenge] = useState<any>([]);
 
   const { response, error, isLoading } = useFetch(CHALLENGES_URL);
-
+  const { response: user } = useFetch(USER_SELF);
```

Comment:
Two responses?
And we show all the challenges on this page, there is another page for personal challenges.

Reply:
when the user subscribes to the challenges we are passing the userId as data. So as to get userId I am calling users/self/

Reply:
The second `useFetch` call will overwrite the `response` constant of the first `useFetch` call.
A good approach will be, writing a separate function to get the details of the logged-in user, and use that in a `useEffect` to get the details at the component mounting phase.